onboarding screen added

harshvardhan-takawale opened the pull request:
Fixes #421 

Changes:
Introduction screens added in viewpager.

harshvardhan-takawale commented:
@brijeshshah13 @mhutti1 Please review. I have created simple viewpager type of onboarding activity.
I don't know why Travis CI is taking so long.

harshvardhan-takawale commented:
![capture1](https://user-images.githubusercontent.com/28982436/36564172-6e69f1ba-1842-11e8-9073-50641ba1c05d.PNG)
![capture2](https://user-images.githubusercontent.com/28982436/36564181-787c0738-1842-11e8-8e2b-d64c66add031.PNG)
![capture3](https://user-images.githubusercontent.com/28982436/36564186-7d176648-1842-11e8-949e-d3176e670877.PNG)
![capture4](https://user-images.githubusercontent.com/28982436/36564201-82e24994-1842-11e8-8fda-b10d265b8f97.PNG)
![capture5](https://user-images.githubusercontent.com/28982436/36564207-8686af0e-1842-11e8-9bde-f34524aa8f53.PNG)

srv-twry commented:
@harshvardhan-takawale The UI looks good but can you please attach a GIF instead of these screenshots in order to get a better feel of what is happening. Thanks.

harshvardhan-takawale commented:
@srv-twry Sure. Here you go.
![20180225_153525](https://user-images.githubusercontent.com/28982436/36640327-30da9be0-1a42-11e8-815e-30e79e7ebc2f.gif)

mhutti1 commented:
I would put skip in the middle and use arrows/next previous either side of it.

harshvardhan-takawale commented:
@brijeshshah13 @mhutti1 I have added the "previous button" and changed the code according to proper indentation. I came to know now that there was a short-cut for doing this.

harshvardhan-takawale commented:
@mhutti1 @brijeshshah13 @srv-twry any more suggestions?

brijeshshah13 commented:
@mhutti1 LGTM. Verified on API 25 device and API 16 emulator.

mhutti1 commented on app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java:
```diff
@@ -429,6 +429,17 @@ public void onCreate(Bundle savedInstanceState) {
     ActionBarDrawerToggle drawerToggle = new KiwixActionBarDrawerToggle(this, drawerLayout, toolbar);
 
+    Boolean isFirstRun = getSharedPreferences("PREFERENCE", MODE_PRIVATE)
+            .getBoolean("isFirstRun", true);
```
4 Space line continuations

mhutti1 commented on app/src/main/java/org/kiwix/kiwixmobile/WelcomeActivity.java:
```diff
@@ -0,0 +1,164 @@
+  }
+
+  private void addBottomDots(int position) {
+
```
no random lines like this

mhutti1 commented on app/src/main/res/layout/welcome_detail1.xml:
```diff
@@ -0,0 +1,37 @@
+      android:gravity="center_horizontal"
+      android:orientation="vertical">
+
+        <ImageView
```
If im counting right this is 4 spaces.

harshvardhan-takawale commented:
@mhutti1 I apologize for repeating these mistakes. Please review.

mhutti1 commented on app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java:
```diff
@@ -429,6 +429,17 @@ public void onCreate(Bundle savedInstanceState) {
 
     ActionBarDrawerToggle drawerToggle = new KiwixActionBarDrawerToggle(this, drawerLayout, toolbar);
 
+    Boolean isFirstRun = getSharedPreferences("PREFERENCE", MODE_PRIVATE)
```
We have a new shared preference system now.

mhutti1 replied:
actually my bad about this. It isn't in just yet. Keeping track of 25 pull requests is getting a bit out of hand :P

mhutti1 commented on app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java:
```diff
@@ -429,6 +429,17 @@ public void onCreate(Bundle savedInstanceState) {
+
+
+    getSharedPreferences("PREFERENCE", MODE_PRIVATE).edit()
+            .putBoolean("isFirstRun", false).commit();
```
4 space tab.

mhutti1 commented on app/src/main/java/org/kiwix/kiwixmobile/WelcomeActivity.java:
```diff
@@ -0,0 +1,144 @@
+        }
+      }
+    });
+
```
Please remove random whitespace such as this.

mhutti1 commented on app/src/main/java/org/kiwix/kiwixmobile/WelcomeActivity.java:
```diff
@@ -0,0 +1,144 @@
+
+    @Override
+    public void onPageSelected(int position) {
+
```
And this.

harshvardhan-takawale commented:
@mhutti1 I have made the changes and rebased.

harshvardhan-takawale commented:
@mhutti1 Any reviews for this?

srv-twry commented:
@harshvardhan-takawale We now have a SharedPreferenceManager for managing default shared preferences. Please use that :smile:

harshvardhan-takawale commented:
@srv-twry Done.

srv-twry commented on app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java:
```diff
@@ -425,6 +426,15 @@ public void onCreate(Bundle savedInstanceState) {
 
     ActionBarDrawerToggle drawerToggle = new KiwixActionBarDrawerToggle(this, drawerLayout, toolbar);
 
+    SharedPreferences firstRunPref = PreferenceManager.getDefaultSharedPreferences(this);
```
Please check the newly created ```SharedPreferenceManager``` and use it.

harshvardhan-takawale replied:
SharedPreferenceUtil?

srv-twry replied:
Yes. Sorry for late reply

srv-twry commented on app/src/main/java/org/kiwix/kiwixmobile/WelcomeActivity.java:
```diff
@@ -0,0 +1,141 @@
+    prev = (Button) findViewById(R.id.btn_prev);
+    prev.setVisibility(View.GONE);
+
+
```
Don't add too many extra lines.

srv-twry commented on app/src/main/java/org/kiwix/kiwixmobile/WelcomeActivity.java:
```diff
@@ -0,0 +1,141 @@
+      }
+    });
+
+    prev.setOnClickListener(new View.OnClickListener() {
```
Let's try to keep ```onCreate``` short. Please create another method ```setOnClickListeners``` and extract this code from here and just call it from the ```onCreate```.

srv-twry commented on app/src/main/java/org/kiwix/kiwixmobile/WelcomeActivity.java:
```diff
@@ -0,0 +1,141 @@
+
+    @Override
+    public void onPageScrollStateChanged(int state) {
+
```
If there is nothing to do, then please comment either ```// No implementation``` or ```// Nothing to do```

srv-twry commented on app/src/main/java/org/kiwix/kiwixmobile/WelcomeActivity.java:
```diff
@@ -0,0 +1,141 @@
+  };
+
+
+  public class ViewPagerAdapter extends PagerAdapter {
```
Can you please try and move it to a new file ? It helps keeps the code organised and activities shorter.

srv-twry commented on app/src/main/res/layout/welcome_detail1.xml:
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
```RelativeLayout``` is outdated in the favour of ```ConstraintLayout```. ConstraintLayouts are extremely efficient in performance. If possible, convert it to ConstraintLayout.

harshvardhan-takawale commented:
@srv-twry @mhutti1 Please review.

ISNIT0 commented:
Nice stuff Harshvardhan.
My one point is that the icons/words shown in the introduction don't seem to map to the actual behaviour in the app.
E.g. "Get Some Knowledge":
- The icon isn't used in the actual app/on the landing page
- The words aren't the same as in the app
So for that specific example, it may be worth adding some text underneath saying "Tap Get Content to start downloading" or something?

- "Get files for different fields" doesn't really make a huge amount of sense - what do you mean here?

Just some thoughts, but it's looking great!

harshvardhan-takawale commented:
@ISNIT0 Thanks! Actually, when I made this, I kept the text and the images as a placeholder and thought I would change it according to the reviews of others. Also, are there any other things that you would like to change in the functionality?

ISNIT0 commented:
Nothing in particular, but it may be worth posting a new GIF since your button changes.

harshvardhan-takawale commented:
@ISNIT0 Any suggestion for the images I should use?

ISNIT0 commented:
@harshvardhan-takawale I would recommend only images/icons actually within the app (you could always add the to the app too).

E.g. the download page, could have that cloud-dl icon, and you could add the same to the "Get Content" button.

brijeshshah13 commented:
@harshvardhan-takawale Please post a new GIF of the onboarding screen.

abdulwd commented:
Moved to #754
